modules/xfem/src/userobjects/InterfaceMeshCutUserObjectBase.C

```diff
@@ -0,0 +1,199 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceMeshCutUserObjectBase.h"
+#include "XFEMMovingInterfaceVelocityBase.h"
+
+InputParameters
+InterfaceMeshCutUserObjectBase::validParams()
+{
+  InputParameters params = GeometricCutUserObject::validParams();
+  params.addRequiredParam<MeshFileName>("mesh_file", "Mesh file for the XFEM geometric cut.");
+  params.addParam<UserObjectName>("interface_velocity_uo",
+                                  "The name of userobject that computes the velocity.");
+  params.addParam<FunctionName>("interface_velocity_function",
+                                "The name of function that provides interface velocity.");
+  params.addParam<bool>("output_exodus", true, "Whether to output exodus file of cutter mesh.");
+  params.addParam<CutSubdomainID>(
+      "negative_id", 0, "The CutSubdomainID corresponding to a non-positive signed distance");
+  params.addParam<CutSubdomainID>(
+      "positive_id", 1, "The CutSubdomainID corresponding to a positive signed distance");
+  return params;
+}
+
+InterfaceMeshCutUserObjectBase::InterfaceMeshCutUserObjectBase(const InputParameters & parameters)
+  : GeometricCutUserObject(parameters),
+    _interface_velocity(nullptr),
+    _func(nullptr),
+    _mesh(_subproblem.mesh()),
+    _exodus_io(nullptr),
+    _explicit_system(nullptr),
+    _equation_systems(nullptr),
+    _output_exodus(getParam<bool>("output_exodus")),
+    _negative_id(getParam<CutSubdomainID>("negative_id")),
+    _positive_id(getParam<CutSubdomainID>("positive_id"))
+{
+  MeshFileName xfem_cutter_mesh_file = getParam<MeshFileName>("mesh_file");
+  _cutter_mesh = std::make_shared<ReplicatedMesh>(_communicator);
+  _cutter_mesh->read(xfem_cutter_mesh_file);
+  _exodus_io = std::make_unique<ExodusII_IO>(*_cutter_mesh);
```

Comment:
Does this mean that the cutter mesh has to be an exodus file, or can we create an exodus output based on whatever mesh format we read in?

My understanding was that the `MeshCut3DUserObject` that we currently have only works with XDA, but maybe that's not the case. I'd like to make sure they're consistent with each other.

Reply:
No, the cutter mesh can be only type that MOOSE supports, including exodus, XDA, etc. This should be also true for `MeshCut3DUserObject`.  

We only support to output exodus file.